src/ad-cid.js

```diff
@@ -29,13 +29,26 @@ export function getAdCid(adElement) {
   if (!config || !config.clientIdScope) {
     return Promise.resolve();
   }
-  const cidPromise = cidForDocOrNull(adElement.getAmpDoc()).then(cidService => {
+  return getOrCreateAdCid(adElement.getAmpDoc(), adElement.win,
+                          config.clientIdScope, config.clientIdCookieName);
+}
+
+/**
+ * @param {!./service/ampdoc-impl.AmpDoc} ampDoc
+ * @param {!Window} win
+ * @param {!string} clientIdScope
+ * @param {string=} opt_clientIdCookieName
+ * @return {!Promise<string|undefined>} A promise for a CID or undefined.
+ */
+export function getOrCreateAdCid(
+    ampDoc, win, clientIdScope, opt_clientIdCookieName) {
```

Comment:
yes, it should exist: https://github.com/ampproject/amphtml/blob/master/src/service/ampdoc-impl.js#L186